Yoast/wordpress-seo
# Move wpseo-links-table-not-accessible notification to site health

## Context


* We want to move notifications to the WordPress Site Health

## Summary


This PR can be summarized in the following changelog entry:

* Moves the text link counter notification from the SEO Dashboard to the WordPress' Site Health.


## Relevant technical choices:

* added an upgrade routine for 14.1-RC0


## Test instructions
This PR can be tested by following these steps:

**Make the old notification appear in the Notification Center**
- make sure the text link counter feature is enabled under SEO > General > Features
- add `return false;` at the first line of the `check_table()` method within `admin/links/class-link-table-accessible.php`
- **alternatively**, you can delete the `wp_yoast_seo_links` table from the database
- go to SEO dashboard and see the old notification displayed:

<img width="962" alt="Screenshot 2020-04-20 at 11 35 45" src="https://user-images.githubusercontent.com/1682452/79742066-8f906680-8302-11ea-8e94-055ddf703fb7.png">


**Switch to this branch and build**

### Test not passed

- go to the WordPress Site Health page
- see the test result is within the "recommended improvements" section:

<img width="892" alt="Screenshot 2020-04-15 at 14 13 21" src="https://user-images.githubusercontent.com/1682452/79338205-2cc85500-7f27-11ea-8ed2-4df250712a71.png">

- you may want to repeat the test by doing the same thing for the `wp_yoast_seo_meta` table and `admin/class-meta-table-accessible.php` but I'd consider this a redundant check

The copy is:
> The text link counter feature is not working as expected
> For this feature to work, Yoast SEO needs to create a table in your database. We were unable to create this table automatically.
> [link]Find out how to solve this problem on our knowledge base.[/link]


- trigger the upgrade routine by using the Yoast Test Helper plugin
- from the root of your Docker environment, delete the related WordPress transient by running `./wp.sh transient delete wpseo_link_table_inaccessible`
- please notice you will need to delete the transient each time you change whether the database tables are accessible or not
- verify the old notification on SEO > General > Notifications, doesn't appear any longer

### Test passed 
- remote the `return false` added at the first line of the `check_table()` method
- if you deleted the database table, restore it by adding this code somewhere in a place where it is executed:
```
$link_installer = new WPSEO_Link_Installer();
$link_installer->install();
```
- remove the above code
- go to the WordPress Site Health page
- expand the Passed tests section
- see the test result within the section

<img width="844" alt="Screenshot 2020-04-15 at 14 14 40" src="https://user-images.githubusercontent.com/1682452/79337721-55038400-7f26-11ea-80af-9c0a42438932.png">

The copy is:
> The text link counter is working as expected
> The text link counter helps you improve your site structure. [link]Find out how the text link counter can enhance your SEO[/link].


### Test with the feature disabled
- go to SEO > General > Features
- disable the text link counter feature
- go to the Site Health page
- see the test didn't run


## UI changes

## Documentation

## Quality assurance


Fixes https://github.com/Yoast/wordpress-seo/issues/14494
